core/src/test/java/org/apache/accumulo/core/util/GuavaCacheCounterUtil.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.util;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.cache.Cache;
+
+/**
+ * Unit test helper for counting guava cache entries. Guava cache size() is approximate, and can
+ * include garbage-collected entries, so we iterate to get the actual cache size
+ */
+public class GuavaCacheCounterUtil {
+  private static final Logger LOG = LoggerFactory.getLogger(GuavaCacheCounterUtil.class);
+  private static final int retries = 5;
+  private static final long delayMills = 1_000;
+
+  /**
+   * guava cache size() is approximate, and can include garbage-collected entries, so we iterate to
+   * get the actual cache size
+   */
+  public static long cacheCount(final Cache<?,?> cache) {
+    return cache.asMap().entrySet().stream().count();
```

Comment:
```suggestion
    return cache.asMap().entrySet().size();
```

Reply:
The size method was explicitly avoided (as noted in the comment) because of the way guava handles size() with deleted / gc eligible entries

Reply:
Oh gotcha, I was thinking that applies to `cache.size()` as opposed to `cache.asMap().entrySet().size()`.